packages/transaction-controller/src/TransactionController.ts

```diff
@@ -1857,6 +1871,30 @@ export class TransactionController extends BaseController<
     return this.getTransaction(txId);
   }
 
+  /**
+   * Update the isFocus state of a transaction.
+   *
+   * @param transactionId - The ID of the transaction to update.
+   * @param isFocused - The new focus state.
+   */
+  updateTransactionFocus(transactionId: string, isFocused: boolean) {
+    const transactionMeta = this.getTransaction(transactionId);
+
+    if (!transactionMeta) {
+      throw new Error(`Transaction with id ${transactionId} not found`);
+    }
+
+    const updatedTransactionMeta = {
+      ...transactionMeta,
+      isFocused,
+    };
+
+    this.updateTransaction(
```

Comment:
We want to exclusively use `#updateTransactionInternal` going forward as it uses callbacks to update meaning there is no risk of race conditions from multiple updates.

It would also mean we don't need to query the existing state in this method.